src/app/page.tsx

```diff
@@ -1,41 +1,37 @@
-import {
-	EstimatedPackage,
-	PackageOwnership,
-	tideliftMeUp,
-} from "tidelift-me-up";
-
 import { Footer } from "~/components/Footer";
 import { MainArea } from "~/components/MainArea";
 import { OptionsForm } from "~/components/OptionsForm";
 import { ResultDisplay } from "~/components/ResultDisplay";
 import { ScrollButton } from "~/components/ScrollButton";
+import { SearchParamsType, fetchData } from "~/utils/fetchData";
 
+import { metadata as defaultMetadata } from "./layout";
 import styles from "./page.module.css";
 
 export interface HomeProps {
-	searchParams: Record<string, unknown>;
+	searchParams: SearchParamsType;
 }
 
-export default async function Home({ searchParams }: HomeProps) {
-	const options = {
-		ownership: undefinedIfEmpty(
-			[
-				searchParams["author"] === "on" && "author",
-				searchParams["maintainer"] === "on" && "maintainer",
-				searchParams["publisher"] === "on" && "publisher",
-			].filter(Boolean) as PackageOwnership[],
-		),
-		since: (searchParams["since"] || undefined) as string | undefined,
-		username: searchParams.username as string,
-	};
-	let result: Error | EstimatedPackage[] | undefined;
+export async function generateMetadata({ searchParams }: HomeProps) {
+	const { options, result } = await fetchData(searchParams);
+	const username = options.username || "";
+	const packageCount = Array.isArray(result) ? result.length : 0;
 
-	try {
-		result = options.username ? await tideliftMeUp(options) : undefined;
-	} catch (error) {
-		result = error as Error;
+	if (!username) {
```

Comment:
[Refactor] I think it'd be nice to do this _"if no username, return default metadata"_ check earlier on. We can avoid the overhead of calling to an async function if we parse options first and bail out otherwise.

I know `await fetchData(...)` right now is super fast, but just conceptually an async tick is nice to avoid - and these api helpers tend to get slower over time.

How about:

1. Also making `getOptions` available to this file (either from `fetchData.ts` or some `getOptions.ts`/`options.ts` file)
2. Having the code do the `options.username` check first
3. Then only call to `fetchData` after that